src/libmeasurement_kit/dns/getaddrinfo_async.hpp

```diff
@@ -0,0 +1,121 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/dns.hpp>
+
+namespace mk {
+namespace dns {
+
+#define GETADDRINFO_ASYNC_MAP_ERROR                                            \
+    XX(0, NoError)                                                             \
+    XX(EAI_AGAIN, TemporaryFailureError)                                       \
+    XX(EAI_BADFLAGS, InvalidFlagsValueError)                                   \
+    XX(EAI_FAIL, NonRecoverableFailureError)                                   \
+    XX(EAI_FAMILY, NotSupportedAIFamilyError)                                  \
+    XX(EAI_MEMORY, MemoryAllocationFailureError)                               \
+    XX(EAI_NONAME, HostOrServiceNotProvidedOrNotKnownError)                    \
+    XX(EAI_OVERFLOW, ArgumentBufferOverflowError)                              \
+    XX(EAI_SERVICE, NotSupportedServnameError)                                 \
+    XX(EAI_SOCKTYPE, NotSupportedAISocktypeError)
```

Comment:
It seems like you are missing some of the possible error values for `getaddrinfo`, namely:

* `EAI_NODATA `: "No address associated with NAME."
* `EAI_ADDRFAMILY `: "Address family for NAME not supported"
* `EAI_SYSTEM`: "System error returned in `errno`"